codefresh/resource_pipeline.go

```diff
@@ -1084,16 +1108,35 @@ func mapResourceToPipeline(d *schema.ResourceData) (*cfclient.Pipeline, error) {
 			OriginalYamlString: originalYamlString,
 		},
 		Spec: cfclient.Spec{
-			PackId:                       d.Get("spec.0.pack_id").(string),
-			RequiredAvailableStorage:     d.Get("spec.0.required_available_storage").(string),
-			Priority:                     d.Get("spec.0.priority").(int),
-			Concurrency:                  d.Get("spec.0.concurrency").(int),
-			BranchConcurrency:            d.Get("spec.0.branch_concurrency").(int),
-			TriggerConcurrency:           d.Get("spec.0.trigger_concurrency").(int),
-			PermitRestartFromFailedSteps: d.Get("spec.0.permit_restart_from_failed_steps").(bool),
+			PackId:                   d.Get("spec.0.pack_id").(string),
+			RequiredAvailableStorage: d.Get("spec.0.required_available_storage").(string),
+			Priority:                 d.Get("spec.0.priority").(int),
+			Concurrency:              d.Get("spec.0.concurrency").(int),
+			BranchConcurrency:        d.Get("spec.0.branch_concurrency").(int),
+			TriggerConcurrency:       d.Get("spec.0.trigger_concurrency").(int),
 		},
 	}
 
+	hasPermitRestartChanged := d.HasChange("spec.0.permit_restart_from_failed_steps")
```

Comment:
@masontikhonov HasChange functions are normally used in custom diff functions. We should instead just check  if `permit_restart_from_failed_steps_use_account_settings` is set to true and in such case set the property to nil

Reply:
Fixed